internal/api/rbac/roles.go

```diff
@@ -579,6 +587,13 @@ func (r *rolesDatabase) Update(
 	replaceClaimAnnotation(sa, rbacapi.AnnotationKeyOIDCSubjects, kargoRole.Subs)
 	replaceClaimAnnotation(sa, rbacapi.AnnotationKeyOIDCEmails, kargoRole.Emails)
 	replaceClaimAnnotation(sa, rbacapi.AnnotationKeyOIDCGroups, kargoRole.Groups)
+	if annotations := kargoRole.GetAnnotations(); annotations != nil {
```

Comment:
This check is also redundant, `delete` can safely operate on `sa.Annotations` (as long as it's a map, but it's allowed to be a `nil` map).